common.py

```diff
@@ -400,54 +399,77 @@ def prepare_filelist(self, names, path, goto, indent):
         items += files
         return items
 
-    def is_hidden(self, filename, path, goto=''):
+    def is_hidden(self, item, path, goto='', get_is_dir=False, dirs_only=False):
         if not (path or goto):  # special case for ThisPC
             return False
-        tests = self.view.settings().get('dired_hidden_files_patterns', ['.*'])
-        if isinstance(tests, str):
-            tests = [tests]
-        if any(fnmatch.fnmatch(filename, pattern) for pattern in tests):
-            return True
-        if sublime.platform() != 'windows':
-            return False
-        # check for attribute on windows:
-        try:
-            attrs = ctypes.windll.kernel32.GetFileAttributesW(join(path, goto, filename))
-            assert attrs != -1
-            result = bool(attrs & 2)
-        except (AttributeError, AssertionError):
-            result = False
-        return result
-
-    def try_listing_directory(self, path):
+        show_hidden = self.show_hidden
+        show_excluded = self.view.settings().get('dired_show_excluded_files', True)
+        is_hidden = False
+        filename = item['file']
+        fullpath = join(path, goto, filename)
+        is_dir = item['is_dir']
+        result = lambda: is_hidden if not get_is_dir else [is_hidden, is_dir]
+        if dirs_only and not is_dir:
+            return result()
+        if not is_hidden and not show_excluded:
+            if is_dir:
+                tests = self.view.settings().get('folder_exclude_patterns', [])
+                if any(fnmatch.fnmatch(filename, p) for p in tests):
+                    is_hidden = True
+            else:
+                tests = self.view.settings().get('file_exclude_patterns', [])
+                if any(fnmatch.fnmatch(filename, p) for p in tests):
+                    is_hidden = True
+        if not is_hidden and not show_hidden:
+            tests = self.view.settings().get('dired_hidden_files_patterns', ['.*'])
+            if isinstance(tests, str):
+                tests = [tests]
+            if any(fnmatch.fnmatch(filename, p) for p in tests):
+                is_hidden = True
+            if not is_hidden and NT:
+                # check for attribute on windows:
+                try:
+                    attrs = ctypes.windll.kernel32.GetFileAttributesW(fullpath)
+                    assert attrs != -1
+                    if bool(attrs & 2):
+                        is_hidden = True
+                except:
+                    pass
+        return result()
+
+    def listdir(self, path):
+        return [{'file': file, 'is_dir': isdir(join(path, file))} for file in os.listdir(path)]
+
+    def listdir_only_dirs(self, path):
+        return [item for item in self.listdir(path) if item["isdir"] == True]
+
```

Comment:
These methods now return directory list. They are the only ones that do `isdir` check so no other methods need to.